Realm/ObjectServerTests/RLMObjectServerPartitionTests.mm

```diff
@@ -0,0 +1,108 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMSyncTestCase.h"
+
+#import "RLMApp_Private.hpp"
+
+#if TARGET_OS_OSX
+
+#pragma mark - Helpers
+
+// These are defined in Swift. Importing the auto-generated header doesn't work
+// when building with SPM, so just redeclare the bits we need.
+@interface RealmServer : NSObject
++ (RealmServer *)shared;
+- (NSString *)createAppForBSONType:(NSString *)bsonType
+                             error:(NSError **)error;;
+@end
+
+#pragma mark ObjectServer Partition Tests
+
+@interface RLMObjectServerPartitionTests : RLMSyncTestCase
+@end
+
+@implementation RLMObjectServerPartitionTests
+
+- (void)roundTripForPartitionValue:(id<RLMBSON>)value  {
+    NSString *appId;
+    if (self.isParent) {
+        NSError *error;
+        appId = [RealmServer.shared createAppForBSONType: [self partitionBsonType:value]  error:&error];
+
+        if (error) {
+            XCTFail(@"Could not create app for partition value %@d", value);
```

Comment:
You'll want to also return after this so that it doesn't try to run the actual test.